util/PostgresMigrations/HelperScripts/2024-08-26_00_FinalFlexibleCollectionsDataMigrations.psql

```diff
@@ -0,0 +1,35 @@
+﻿-- Migrate Custom users who only have 'editAssignedCollections' and/or 'deleteAssignedCollections' custom permissions to the User type.
+UPDATE "OrganizationUser"
+SET
+    "Type" = 2,
+    "Permissions" = NULL
+WHERE
+    "Type" = 4
+    AND "Permissions" IS NOT NULL
+    AND "Permissions" ~ '^\s*\{.*\}\s*$'  -- Check if Permissions is a valid JSON object
```

Comment:
This is rough. But I see that postgres didn't support Is_Json until [relatively recently](https://modern-sql.com/caniuse/is-json) and I don't think we've defined any version support for our EF databases.

Looking at what we've done previously in Postgres, I found this: https://github.com/bitwarden/server/blob/main/util/PostgresMigrations/HelperScripts/2021-09-21_01_SplitManageCollectionsPermission.psql#L1. In that case it doesn't look like we guarded against invalid json at all, so this already seems better as a rough check.

I think this is a fine solution for now.